Lesson 04/news_reader/index.js

```diff
@@ -0,0 +1,269 @@
+					url: $title.attr('href').trim().replace(/^\//, app.siteSettings.vesti.articleUrl)
+				});
+			});
+		}
+		
+		clientRes.render('news', {
+			title: app.viewSettings.pageTitle,
+			mainPage: app.viewSettings.mainPage,
+			newsSite: app.siteSettings.vesti.title,
+			newsCount: newsCount,
+			newsCategory: category,
+			news: news
+		});
+	});
+}
+
+// News handler.
+app.post('/news', urlencodedParser, (clientReq, clientRes) => {
+	switch (clientReq.body.newsSite) {
+		case 'vesti': fetchNewsVesti(clientReq, clientRes); break;
+	}
+});
+
+
+// News categories handler.
+app.get('/news/:site', (clientReq, clientRes) => {
+	const newsSite = app.siteSettings[clientReq.params.site];
+	
+	if (!newsSite) {
+		responseData.result = 404;
+		clientRes.send(JSON.stringify(responseData));
+		
+		return;
+	}
+	
+	request(newsSite.requestUrl, (err, res, body) => {
+		if (err) {
+			responseData.result = 1;
+			responseData.message = err.message;
+		} else if (res.statusCode !== 200) {
+			responseData.result = res.statusCode;
+			responseData.message = res.statusText;
+		}
+		
+		if (responseData.result) {
+			clientRes.send(JSON.stringify(responseData));
+			
+			return;
+		}
+		
+		switch (newsSite.name) {
+			case 'vesti': responseData.data = parseCategoryVesti(cheerio.load(body)); break;
+			default: break;
+		}
+
+		if (!responseData.data) {
+			responseData.result = 1;
+		}
+		
+		clientRes.send(JSON.stringify(responseData));
+	});
+});
+
+
+// Root request handler.
+app.get('/', (req, res) => {
+	const countOptions = [
+		{value: 10, text: '10'},
+		{value: 20, text: '20'},
+		{value: 30, text: '30'},
+		{value: 0, text: 'All'}
+	];
+	let selected;
+	
+	// Parse newsCount param.
+	if (req.query.newsCount) {
+		req.query.newsCount = parseInt(req.query.newsCount.trim());
+		
+		countOptions.forEach((elem) => {
+			if (elem.value === req.query.newsCount) {
+				elem.selected = true;
+				selected = true;
+			}
+		});
+	}
+
+	// Default select is 20 news count.
+	if (!selected) {
+		countOptions[1].selected = true;
+	} else {
+		selected = undefined;
+	}
+	
+	// Parse newsSite param.
+	if (req.query.newsSite) {
+		req.query.newsSite = req.query.newsSite.trim();
+		
+		for (const site in app.siteSettings) {
+			if (app.siteSettings[site].name === req.query.newsSite) {
+				app.siteSettings[site].selected = true;
+				selected = true;
+			}
+		}
+	}
+		
+	// Default select is the first site.
+	if (!selected) {
+		app.siteSettings[0].selected = true;
+	} else {
+		selected = undefined;
+	}
+	
+	res.render('newsFlash', {
+		newsPage: app.viewSettings.newsPage,
+		title: app.viewSettings.title,
+		countOptions: countOptions,
+		newsSites: app.siteSettings
+	});
+});
+
+
+// Styles/scripts folder request handlers.
+function sendFile(res, path, fileName) {
+	if (fs.existsSync(path + fileName)) {
+		res.sendFile(path + fileName);
+	} else {
+		renderDefaultPage(res);
+	}
+}
+
+app.get('/styles/:file', (req, res) => {
```

Comment:
app.use(express.static('./<path>'))

Reply:
Doesn't work in this form.
Working solution for me is:
```javascript
app.use('/styles', express.static('styles'));
app.use('/scripts', express.static('scripts'));
```